code/job_dispatch/run_capsule.py

```diff
@@ -0,0 +1,159 @@
+"""
+Generates the input analysis model from the user provided query
+"""
+
+import argparse
+import json
+import logging
+import numpy as np
+import pathlib
+
+from job_dispatch import utils
+from job_dispatch.analysis_input_model import InputAnalysisModel
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+
+def get_input_parser() -> argparse.ArgumentParser:
```

Comment:
What do you think about making this a `BaseSettings` model? It would offer validation, type hinting and documentation if users are using this from code